charts/metering-ansible-operator/templates/crds/reportquery.crd.yaml

```diff
@@ -23,3 +23,104 @@ spec:
     type: date
     JSONPath: .metadata.creationTimestamp
+
+  validation:
+    openAPIV3Schema:
+      type: object
+      required:
+      - spec
+      properties:
+        spec:
+          type: object
+          required:
+          - columns
+          - query
+          properties:
+            columns:
+              type: array
+              minItems: 1
+              items:
+                type: object
+                required:
+                - name
+                - type
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  type:
+                    type: string
+                    enum:
+                      - BOOLEAN
+                      - TINYINT
+                      - SMALLINT
+                      - INTEGER
+                      - BIGINT
+                      - REAL
+                      - DOUBLE
+                      - DECIMAL
+                      - VARCHAR
+                      - CHAR
+                      - VARBINARY
+                      - JSON
+                      - DATE
+                      - TIME
+                      - TIMESTAMP
+                      - ARRAY
+                      - MAP
+                      - MAP<VARCHAR, VARCHAR>
+                      - MAP<VARCHAR, INT>
+                      - MAP<INT, INT>
+                      - MAP<INT, VARCHAR>
+                      - ROW
+                      - IPADDRESS
+                      - UUID
+                      - HYPERLOGLOG
+                      - P4HYPERLOGLOG
+                      - QDIGEST
+                  unit:
+                    type: string
+                    enum:
+                      - date
+                      - kubernetes_pod
+                      - kubernetes_persistentvolumeclaim
+                      - kubernetes_persistentvolume
+                      - kubernetes_storageclass
+                      - kubernetes_namespace
+                      - bytes
+                      - byte_seconds
+                      - time
+                      - cpu_core_seconds
+                      - cpu_cores
+                      - memory_bytes
+                      - memory_bytes_seconds
+                      - seconds
+                  tableHidden:
+                    type: boolean
+            inputs:
+              type: array
+              minItems: 1
+              items:
+                type: object
+                required:
+                - name
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  type:
+                    type: string
+                    enum:
+                      - string
+                      - integer
+                      - time
+                      - ReportDataSource
+                      - ReportQuery
+                      - Report
+                  required:
+                    type: boolean
```

Comment:
This also looks misplaced

Reply:
'required` isn't saying that a field is required, but rather the name of the field. So it's `inputs.required` rather than something that is required. Tricky.

Reply:
Ah right